### ISSUE TITLE
Calculating cross entropy from scratch

### ISSUE THREAD

**Issue by ibab** [collinsnji/ML#1]
Here's the line that calculates the cross entropy in your logistic regression script:
https://github.com/collinsnji/ML/blob/5d32c7fd3a0229f0393c84566e74c1520588a284/logistic_regression.py#L15

If you want to do this yourself, it's not too difficult.
You can use `tf.nn.softmax` applied to the output of the network to get probabilities `probs`.
To pick out the right elements for each examples, you can make use of the following trick:
Using the `labels`, you can can get a so-called "one-hot" tensor, which gives you a `1` when the index is equal to the label and `0` else.
If you then multiply this to the log(p), it will pick out the right elements:
```python
one_hot = tf.one_hot(labels, 3)  # the 3 is the number of classes in the dataset.
cross_entropies = tf.reduce_sum(one_hot * tf.log(probs), axis=-1)
loss = -tf.reduce_mean(cross_entropies)
```

Give this a try and check that the output matches the original version of the script :)

**Comment by ibab** [comment]
Looks good :+1:
